src/main/java/dev/aziz/grocerystore/handlers/PriceHandler.java

```diff
@@ -1,27 +1,52 @@
 package dev.aziz.grocerystore.handlers;
 
 import dev.aziz.grocerystore.entities.BasketItem;
+import dev.aziz.grocerystore.entities.UserPromotion;
+import lombok.extern.slf4j.Slf4j;
 
 import java.math.BigDecimal;
 import java.math.BigInteger;
+import java.time.Instant;
 import java.util.List;
 
+@Slf4j
 public class PriceHandler {
 
+    private final UserPromotion userPromotion;
+
     private List<BasketItem> basketItems;
 
-    public PriceHandler(List<BasketItem> basketItems) {
+    public PriceHandler(UserPromotion userPromotion, List<BasketItem> basketItems) {
+        this.userPromotion = userPromotion;
         this.basketItems = basketItems;
     }
 
     public BigDecimal computePrice() {
+        Long id = basketItems.get(0).getUser().getId();
+        Integer minAmount = userPromotion.getPromotionConfig().getMinimumAmount();
+        Integer freeAmount = userPromotion.getPromotionConfig().getFreeAmount();
+        Instant date = userPromotion.getPromotionConfig().getEndDate();
+
+        int value = Instant.now().compareTo(date);
+        int total = minAmount + freeAmount;
+        int free = 0;
+
         BigDecimal totalPrice = new BigDecimal(BigInteger.ZERO);
-        for (BasketItem basketItem : basketItems) {
-            totalPrice = totalPrice.add(
-                    basketItem.getItem().getPrice().multiply(BigDecimal.valueOf(basketItem.getAmount()))
-            );
+
+        if (value < 0) {
```

Comment:
If the promotion is expired, you still have a price, but the promotion isn't applied.